Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,744 @@
+                                    * self._sampleAndEnvironmentProperties['NMoles'].value,
+                                    OutputWorkspace='norm')
+        if self.getPropertyValue('OutputTreatment') == 'Sum':
+            tmp_name = '{}_1'.format(self.getPropertyValue('OutputWorkspace'))
+            RenameWorkspace(InputWorkspace=mtd[ws][0].name(), OutputWorkspace=tmp_name)
+            to_remove = ['norm']
+            for entry_no in range(1, mtd[ws].getNumberOfEntries()):
+                ws_name = mtd[ws][entry_no].name()
+                Plus(LHSWorkspace=tmp_name, RHSWorkspace=ws_name, OutputWorkspace=tmp_name)
+                to_remove.append(ws_name)
+            # expected total cross-section of unpolarised neutrons in V is 1/3 * sum of all measured c-s,
+            # and normalised to 0.404 barns times the number of moles of V:
+            Divide(LHSWorkspace='norm', RHSWorkspace=tmp_name, OutputWorkspace=tmp_name)
+            DeleteWorkspaces(WorkspaceList=to_remove)
+            GroupWorkspaces(InputWorkspaces=tmp_name, OutputWorkspace=ws)
+        else:
+            if self.getPropertyValue('OutputTreatment') == 'Average':
+                self._merge_polarisations(ws, average_detectors=True)
+            Divide(LHSWorkspace='norm', RHSWorkspace=ws, OutputWorkspace=ws)
+            DeleteWorkspace(Workspace='norm')
+        return ws
+
+    def _set_units(self, ws, process):
+        if process == 'Sample' and self.getPropertyValue('OutputTreatment') in  ['Average','Sum']:
+            self._merge_polarisations(ws, average_detectors=(self.getPropertyValue('OutputTreatment') == 'Average'))
+
+        for entry in mtd[ws]:
+            unit = 'dSigma / dOmega (detector number)'
+            entry.setYUnit(unit)
+            entry.setYUnitLabel(unit)
+        return ws
+
+    def _finalize(self, ws, process, progress):
+        ReplaceSpecialValues(InputWorkspace=ws, OutputWorkspace=ws, NaNValue=0,
+                             NaNError=0, InfinityValue=0, InfinityError=0)
+        mtd[ws][0].getRun().addProperty('ProcessedAs', process, True)
+        RenameWorkspace(InputWorkspace=ws, OutputWorkspace=ws[2:])
+        self.setProperty('OutputWorkspace', mtd[ws[2:]])
+
+    def PyExec(self):
+        process = self.getPropertyValue('ProcessAs')
+        processes = ['Absorber', 'EmptyBeam', 'BeamWithAbsorber', 'Transmission', 'Container', 'Quartz', 'Vanadium', 'Sample']
+        nReports = [3, 2, 2, 3, 3, 3, 10, 10]
+        progress = Progress(self, start=0.0, end=1.0, nreports=nReports[processes.index(process)])
+        ws = '__' + self.getPropertyValue('OutputWorkspace')
+
+        calibration_setting = 'YIGFile'
+        if self.getProperty('InstrumentCalibration').isDefault:
+            calibration_setting = 'None'
+        progress.report('Loading data')
+        Load(Filename=self.getPropertyValue('Run'), LoaderName='LoadILLPolarizedDiffraction',
```

Comment:
Shall we not use LoadAndMerge instead, just in case there could be more than one numor to sum at a given scan step?

Reply:
I think it would be possible, in principle, to use `LoadAndMerge` instead, with some adaptation but right now that way of loading does not reliably work with data that is loaded as `WorkspaceGroup`. The entire group is merged into a single workspace that is broken: trying to open that workspace causes an exception (namely AttributeError: 'Workspace2D' object has no attribute 'getMaxNumberBins'). For now it seems that keeping Load as the loading algorithm is more reliable, also to sum more than one numor per scan step, of course at the price of loss of metadata.